webtau-core/src/main/java/org/testingisdocumenting/webtau/persona/Persona.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 webtau maintainers
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.testingisdocumenting.webtau.persona;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class Persona {
+    private static final ThreadLocal<Persona> currentPersona = new ThreadLocal<>();
```

Comment:
If I understand this right, the thread local is purely to guard against nesting?  I suggest we add a comment explaining that for two reasons:
1. it's a bit different to most of our usage of thread locals which is for keeping and accessing global-ish state
1. it's not immediately obvious why we need it given that all usage of this class is against an explicitly created instance rather than through static functions.

Reply:
ThreadLocal is to be able to have active contexts in a multithreaded execution. So test1 running in thread1 can have its context1 and the same for thread2. 
In my tests I plan to reuse persona instances but it is not required. It is similar to page objects in UI module in a way that you can either pre-create them or create on demand.

Reply:
I forgot about the ability to run tests in parallel.